### PR TITLE
Set ServiceProvider as Customers DID controller

### PR DESCRIPTION
**Pull request by stevenvegt**
New customers will have the Service Providers DID als controller. Currently the Customers DIDs are also self controlled. Is this what we want?

### CODE REVIEW

**Comment by reinkrul** [comment]
Can we give it a more descriptive title and summary?

File: api/api.go [hunk 1]
```diff
@@ -106,7 +108,12 @@ func (w Wrapper) ConnectCustomer(ctx echo.Context) error {
 		town = *connectReq.Town
 	}
 
-	customer, err := w.CustomerService.ConnectCustomer(connectReq.Id, connectReq.Name, town)
+	spID, err := w.SPService.Repository.Get()
+	if err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest, fmt.Errorf("unable to fetch service provider ID: %w", err))
```
Comment (reinkrul) [inline api/api.go:113]: Why fetch just the ID from the bolt? I think `Repository` isn't meant to be exposed (but it was convenient for creating `SPService`). Better call `SPService.Get()` then you know for sure the service provider DID document is resolvable on the network.

File: domain/customers/service.go [hunk 2]
```diff
@@ -1,18 +1,30 @@
...
 	VDRClient  nutsApi.HTTPClient
 	Repository Repository
 }
 
-func (s Service) ConnectCustomer(id, name, town string) (*domain.Customer, error) {
-	didDoc, err := s.VDRClient.Create(nutsApi.DIDCreateRequest{})
+func (s Service) ConnectCustomer(id, name, town string, serviceProviderID *did.DID) (*domain.Customer, error) {
+	selfControl := true
+	var controllers []string
+	if serviceProviderID != nil {
```
Comment (reinkrul) [inline domain/customers/service.go:19]: it's never nil?

File: domain/customers/service.go [hunk 3]
```diff
@@ -1,18 +1,30 @@
...
 
 type Service struct {
 	VDRClient  nutsApi.HTTPClient
 	Repository Repository
 }
 
-func (s Service) ConnectCustomer(id, name, town string) (*domain.Customer, error) {
-	didDoc, err := s.VDRClient.Create(nutsApi.DIDCreateRequest{})
+func (s Service) ConnectCustomer(id, name, town string, serviceProviderID *did.DID) (*domain.Customer, error) {
+	selfControl := true
```
Comment (reinkrul) [inline domain/customers/service.go:17]: shouldn't self-control be false (and capabilityInvocation as well)? Such that the SP manages the DID document?

**Review by reinkrul — CHANGES_REQUESTED** [review]

File: domain/customers/service.go (hunk 2 above)
Comment (stevenvegt) [reply domain/customers/service.go:19]: You might want to create a customer without a controller?

**Pull request merged**